lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -468,29 +476,39 @@ class _Listener {
 
 /// Common functionality that's shared among adapters.
 abstract class _BaseAdapter {
-  _BaseAdapter(
-    this._callback,
-    this.flutterViewElement,
-    this._pointerDataConverter,
-    this._keyboardConverter,
-  ) {
+  _BaseAdapter(this._owner) {
     setup();
   }
 
+  final PointerBinding _owner;
+
+  EngineFlutterView get _view => _owner.view;
+  _PointerDataCallback get _callback => PointerBinding.clickDebouncer.onPointerData;
+  PointerDataConverter get _pointerDataConverter => _owner._pointerDataConverter;
+  KeyboardConverter? get _keyboardConverter => _owner._keyboardConverter;
+
   final List<_Listener> _listeners = <_Listener>[];
-  final DomElement flutterViewElement;
-  final _PointerDataCallback _callback;
-  final PointerDataConverter _pointerDataConverter;
-  final KeyboardConverter _keyboardConverter;
   DomWheelEvent? _lastWheelEvent;
   bool _lastWheelEventWasTrackpad = false;
 
+  DomElement get _viewTarget => _view.dom.rootElement;
+  DomEventTarget get _globalTarget {
+    // When the Flutter app owns the full page, we want to listen on window for
```

Comment:
Isn't returning thew view target implictily covering this case?

Reply:
Nope. The root element is always a `<flutter-view>` element that we insert somewhere. But in the `implicitView` case, we want this to return `window`.